DOCS-57: Fix position & positionCaseInsensitive & positionUTF8 & positionCaseInsensitiveUTF8 functions (#7227)
ClickHouse/ClickHouse: docs/en/query_language/functions/string_search_functions.md

The search is case-sensitive by default in all these functions. There are separate variants for case insensitive search.

## position(haystack, needle) {#position}

> Returns the position (in bytes) of the found substring `needle` in the string `haystack`, starting from 1. Returns 0 if the substring was not found.

Reviewer: why do you repeat parameters and return values twice? (here and below)

Author: Removed

> Works under the assumption that the string contains a set of bytes representing a single-byte encoded text. If this assumption is not met, it returns some result.

Reviewer: both strings?

Author: What?

Reviewer: both arguments

> For a case-insensitive search, use the function [positionCaseInsensitive](#positioncaseinsensitive).

Reviewer: Why do you mention only CaseInsensitive version, but not UTF8?

Author: It hasn't been described yet

Reviewer: why not?

Author: already described, see files

**Syntax**

```sql
position(haystack, needle);
```

Alias: `locate(haystack, needle)`.

**Parameters**

- `haystack` — string, from which substring is to be searched.
- `needle` —  substring, which is to be searched.

**Returned values**

- Starting position (in bytes), if substring was found.
- 0, if the substring was not found.

Type: `Integer`.

**Example**

Query:

```sql
SELECT position('Hello, world!', 'Hello');
```

Result:

```text
┌─position('Hello, world!', 'Hello')─┐
│                                  1 │
└────────────────────────────────────┘
```

## positionCaseInsensitive(haystack, needle) {#positioncaseinsensitive}

> Returns the position (in bytes) of the found substring `needle` in the string `haystack`, starting from 1. Returns 0 if the substring was not found. Use the function for a case-insensitive search. 

Reviewer: * Do we really need this copy-paste?
* What exactly "case-insensitive search" means? It's important to specify if it works only for English letters in ASCII or more than that.

Author: > Do we really need this copy-paste?
- Yes.

> What exactly "case-insensitive search" means? It's important to specify if it works only for English letters in ASCII or more than that.

SELECT positionCaseInsensitive('Привет, мир!', 'мир')

┌─positionCaseInsensitive('Привет, мир!', 'мир')─┐
│                                             15 │
└────────────────────────────────────────────────┘

Reviewer: example is not enough, you need to explain how it works

Author: Write, example, please

Works under the assumption that the string contains a set of bytes representing a single-byte encoded text. If this assumption is not met, it returns some result.

**Syntax**

```sql
positionCaseInsensitive(haystack, needle);
```

**Parameters**

> - `haystack` — string, from which substring is to be searched.

Reviewer: ```suggestion
- `haystack` — string, in which substring will to be searched.
```

Author: Fixed.

> - `needle` —  substring, which is to be searched.

Reviewer: ```suggestion
- `needle` —  substring, which is to be searched.
```
```suggestion
- `needle` —  substring to be searched.
```
P.S. Afaik long dash (—) is not used in English like it is in Russian.

Author: Fixed

Author: I checked it with linguists. A long dash is used in English. Correct the long dash or a colon.

**Returned values**

> - Starting position (in bytes), if substring was found.

Reviewer: important to mention that it starts with 1, not 0

Author: Fixed

- 0, if the substring was not found.

> Type: `Integer`.

Reviewer: Which exactly kind of integers?

Author: Yes

Reviewer: Yes?

Author: What type can still return?
